tools/dissector/ws_common.h

```diff
@@ -11,27 +11,110 @@
 MAJOR * 1000000 + MINOR * 1000 + MICRO
 #define WIRESHARK_VERSION WIRESHARK_VERSION_NUMBER(VERSION_MAJOR,VERSION_MINOR,VERSION_MICRO)
 
+
+#if WIRESHARK_VERSION >= WIRESHARK_VERSION_NUMBER(2, 5, 0)
+// find_conversation after 2.5/2.6 takes endpoint_type instead of port_type
+// but also provides a shortcut function which we will use instead.
+inline conversation_t *ws_find_conversation(packet_info *pinfo) {
+  return find_conversation_pinfo(pinfo, 0);
+}
+#else // if 2.4.x or before
+inline conversation_t *ws_find_conversation(packet_info *pinfo) {
+  return find_conversation(
+    pinfo->fd->num, &pinfo->src, &pinfo->dst,
+    pinfo->ptype, pinfo->srcport, pinfo->destport, 0
+  );
+}
+#endif
+
+#if WIRESHARK_VERSION >= WIRESHARK_VERSION_NUMBER(1,4,0)
+// 1.4 introduced find_or_create_conversation
+inline conversation_t *ws_find_or_create_conversation(packet_info *pinfo) {
+  return find_or_create_conversation(pinfo);
+}
+#else // if 1.3.x or before
+inline conversation_t *ws_find_or_create_conversation(packet_info *pinfo) {
+  conversation_t *conv = NULL;
+  if (!(conv = ws_find_conversation(pinfo))) {
+    conv = conversation_new(
+      pinfo->num, &pinfo->src, &pinfo->dst,
+      pinfo->ptype, pinfo->srcport, pinfo->destport, 0
+    );
+  }
+  return conv;
+}
+#endif
+
+#if WIRESHARK_VERSION >= WIRESHARK_VERSION_NUMBER(2, 2, 0)
+inline guint8* ws_tvb_get_string(wmem_allocator_t* alloc, tvbuff_t *tvb,
+                                 gint offset, gint length)
+{
+  return tvb_get_string_enc(alloc, tvb, offset, length, ENC_ASCII);
+}
+#else
+# define ws_tvb_get_string ::tvb_get_string
+#endif
+
+#if WIRESHARK_VERSION >= WIRESHARK_VERSION_NUMBER(2, 2, 0)
+# define WS_DISSECTOR_RETURN_TYPE int
+# define WS_DISSECTOR_RETURN_VALUE return 0; //TODO
+# define ws_dissector_add_handle dissector_add_for_decode_as
+# define WS_CONV_IDX conv_index
+// create_dissector_handle wants the new dissector_t here
+#define WS_DISSECTOR_EXTRA_PARAM , void*
```

Comment:
This line isn't indented (between the # and "define") like the others in its block (following the #if / #endif structure).  There are a few more like this below, I won't leave comments for each one.